### ISSUE TITLE
Association ignores attributes if serializer is defined

### ISSUE THREAD

**Issue by januszm** [rails-api/active_model_serializers#1948]
TLDR: serialized `relationships` don't include `attributes` if a serializer class is defined for associated model.

#### Expected behavior vs actual behavior

```ruby
class AuthorSerializer < ActiveModel::Serializer
  has_many ::books
end

class BookSerializer < ActiveModel::Serializer
  attribute(:publisher_name) { object.publisher.name }
end
```

In the resulting response for `author`:

```
"relationships": {
  "books": {
    "data": [
      {
        "id": "821",
        "type": "books"
      },
```

We're missing `attributes` section, which we can see when we serialize the associated record directly:

```
"data": {
  "id": "821",
  "type": "books",
  "attributes": {
    "publisher_name": "Some publisher"
```

`attributes` are NOT returned if `BookSerializer` class is defined
if we *remove* `BookSerializer` class, then `attributes` are returned, but they simply contain serialized attributes from the database.

My goal is to define custom attributes like `publisher_name` and return them when rendering parent resource.

#### Environment

ActiveModelSerializers Version *0.10.2*: 

ruby 2.3.1p112 (2016-04-26 revision 54768) [x86_64-darwin15]

Mac OS X 10.11

Integrated application and version *Rails 5.0.0.1*:

**Comment by beauby** [comment]
I suppose you are using the `JsonApi` adapter, in which case this is expected behavior (c.f. the [JSON API spec](http://jsonapi.org)).

**Comment by januszm** [comment]
Thank you . It's confusing because if I omit the associated resource serializer definition (remove BookSerializer in my example) then the response contains attributes section

**Comment by beauby** [comment]
Yeah the behavior you describe is actually a known bug, and it is definitely confusing. If you want nested data, the easiest (but limited and non-standard) solution is to use the `Json` or `Attributes` adapter.

**Comment by januszm** [comment]
Gotcha. Thanks.
